### PR TITLE
Gen 2287 internal balance of with address param

### PR DESCRIPTION
**Pull request by stevenvaleri**
## Motivation

Reduce Periphery Contract Size

## Solution

Move external balanceOf calls all to one internal function.

Alternative approaches tried:
+ External Libraries for Logic - too complicated and more gas
+ Import interfaces instead of full contracts - already handled by compiler
+ move `safeTransfer` to an internal function - increases size
+ have internal balanceOf use an ERC20 as an input param instead of the token address

## Implementer Checklist

All logic remains the same. Refactor to reduce contract size.

Reduced Size:
image

Corrected and Accurate Gas Changes - `balanceOf - address` is this implementation:

image

### CODE REVIEW

**Review by jparklev — APPROVED** [review]
This makes sense and actually reads a little cleaner to me, ty for circling back on this

**Review by fedealconada — APPROVED** [review]
lgtm!

File: pkg/core/src/Periphery.sol [hunk 1]
```diff
@@ -1062,6 +1074,10 @@ contract Periphery is Trust, IERC3156FlashBorrower {
         }
     }
 
+    function _balanceOf(address token, address user) internal returns (uint256 amt) {
+        amt = ERC20(token).balanceOf(user);
+    }
+
```
Comment (IAm0x52) [inline pkg/core/src/Periphery.sol:1080]: Every occurrence of native balanceOf ERC20 method replaced with _balanceOf

**Review by IAm0x52 — COMMENTED** [review]
Fixes look good. Every occurrence of ERC20.balance of changed to internal _balanceOf() function and inputs have been correctly translated

**Pull request merged**